clang/lib/Analysis/UnsafeBufferUsage.cpp

```diff
@@ -420,6 +420,118 @@ AST_MATCHER(CXXConstructExpr, isSafeSpanTwoParamConstruct) {
   return false;
 }
 
+class MaxValueEval : public RecursiveASTVisitor<MaxValueEval> {
+
+  std::vector<llvm::APInt> val;
+  ASTContext &Context;
+  llvm::APInt Max;
+  unsigned bit_width;
+
+public:
+  typedef RecursiveASTVisitor<MaxValueEval> VisitorBase;
+
+  explicit MaxValueEval(ASTContext &Ctx, const Expr *exp) : Context(Ctx) {
+    bit_width = Ctx.getIntWidth(exp->getType());
+    Max = llvm::APInt::getSignedMaxValue(bit_width);
```

Comment:
For the same `bit_width`, the unsigned max value will be different from the signed one, right?  Maybe `Max` should depend on `exp`'s type.

Reply:
Thanks. Will handle this in the next version.

Reply:
Also types of those bitwise operands matter.  For example, `((unsigned long long) -1) & ((unsigned) var)` is of type `unsigned long long` but the value is actually upper-bounded by the max value of `unsigned`.  (https://godbolt.org/z/b1zd1zEje)

Handle this in next version sounds good me.